source/AndroidResolver/src/PlayServicesResolver.cs

```diff
@@ -1975,13 +1995,23 @@ public class AndroidSdkRootChangedArgs : EventArgs {
                 return true;
             };
 
+            Func<ICollection<Dependency>, bool> patchSettingsTemplateGradle = (dependencies) => {
+                if (GradleBuildEnabled && GradleTemplateEnabled &&
+                    SettingsDialogObj.PatchSettingsTemplateGradle) {
+                    return GradleTemplateResolver.InjectSettings(dependencies, out lastError);
+                }
+                return true;
+            };
+
             if (GradleTemplateEnabled &&
                 SettingsDialogObj.PatchMainTemplateGradle) {
                 lastError = "";
                 RunOnMainThread.Run(() => {
-                        finishResolution(GradleTemplateResolver.InjectDependencies(
-                            PlayServicesSupport.GetAllDependencies().Values) &&
-                            patchGradleProperties(), lastError);
+                        var dependencies = PlayServicesSupport.GetAllDependencies().Values;
+                        finishResolution(
+                            GradleTemplateResolver.InjectDependencies(dependencies) &&
+                            patchGradleProperties() &&
+                            patchSettingsTemplateGradle(dependencies), lastError);
```

Comment:
I'm a bit confused by what lastError is meant to do.  Like, if the patchSettingsTemplateGradle changes lastError, I don't think it will affect the variable that is passed into finishResolution (I could be wrong, passing parameters around like this is definitely confusing).

Reply:
I agree this is hard to understand. 

In fact, this actually works (Just tried it). I think this is probably because the function parameters are evaluated in order. Therefore, `patchGradleProperties()` and `patchSettingsTemplateGradle(dependencies)` has already been called before `lastError` is passed to `finishResolution()`.

Reply:
Hmm, I guess that does make sense.  Definitely a bit confusing, but I don't think it is worth blocking for.